### PR TITLE
show error-view instead of yellow notification on content loading errors

### PR DESCRIPTION
**Pull request by ChristophWurst**
implements part 1 of https://github.com/nextcloud/mail/issues/56 – the error view.

Examples:
bildschirmfoto von 2016-09-13 19-30-49
bildschirmfoto von 2016-09-13 19-30-03

@te-online @skjnldsv @jancborchardt how about a nice error-logo above the error message to be consistent with the *no messages in this folder* and *no search results* view?

### CODE REVIEW

**Comment by ChristophWurst** [comment]
Maybe we can also find a better and shorter wording for "Error while loading the selected message" :wink:

**Comment by skjnldsv** [comment]
"Unable to load the desired message" ? :)
Nice idea btw! 👍

**Comment by jancborchardt** [comment]
For the individual folders, I would use the folder-relevant icon for the error icon. For individual mails or when there is no specific icon I’d say let’s use the mail-envelope-icon.

The messages could be a bit improved to this to be less sterile and also a bit fun and helpful. First line is h2 in emptycontent, and second line is the `<p>`

> We couldn’t load the **Inbox** :( (the **Sent mails**, the **Trash** etc.)
> Try reloading the page

> **Vacation planning** isn’t loading :\
> Click on another folder and come back later, maybe?

> There was a problem loading the message :(
> Maybe you deleted it on your phone?

Yes, with the :( smiley. Maybe even cycle randomly between :( and :\ and :/ and ;( – that makes the error message at least a bit fun. ;)

**Comment by ChristophWurst** [comment]
@jancborchardt something like this?
bildschirmfoto von 2016-09-14 21-09-46

We'll need a generic icon for non-special-use folders. Could could make one, please?

**Comment by ChristophWurst** [comment]
> For individual mails or when there is no specific icon I’d say let’s use the mail-envelope-icon.

Ah, didn't read that before…

**Comment by ChristophWurst** [comment]
oh, we have an 'icon-folder' too. Hmm…
bildschirmfoto von 2016-09-14 21-13-26

**Comment by te-online** [comment]
But give the smileys noses, please! 😃
________________________________
Von: Jan-Christoph Borchardt<mailto:notifications@github.com>
Gesendet: ‎14.‎09.‎2016 19:40
An: nextcloud/mail<mailto:mail@noreply.github.com>
Cc: Thomas Ebert<mailto:thomas.ebert@te-online.net>; Mention<mailto:mention@noreply.github.com>
Betreff: Re: [nextcloud/mail] show error-view instead of yellow notification on content loading errors (#57)


For the individual folders, I would use the folder-relevant icon for the error icon. For individual mails or when there is no specific icon I’d say let’s use the mail-envelope-icon.

The messages could be a bit improved to this to be less sterile and also a bit fun and helpful. First line is h2 in emptycontent, and second line is the <p>

We couldn’t load the Inbox :( (the Sent mails, the Trash etc.)
Try reloading the page

Vacation planning isn’t loading :\
Click on another folder and come back later, maybe?

There was a problem loading the message :(
Maybe you deleted it on your phone?

Yes, with the :( smiley. Maybe even cycle randomly between :( and :\ and :/ and ;( – that makes the error message at least a bit fun. ;)

**Comment by jancborchardt** [comment]
@ChristophWurst, yeah, basically like that! :) Is it not possible to bold the foldername instead of using quotes? That would look waaay better.

Also let’s please use the mail icon instead of the folder icon. We do not use the folder icon anywhere in the mail app – it belongs to Files. In fact other mail apps like Gmail use the concept of »Tags« rather than folders so let’s use a generic one.

And below that sentence, put a `<p>` with the second-line text I showed above. Or do you think it’s too much?

**Comment by ChristophWurst** [comment]
> Is it not possible to bold the foldername instead of using quotes? That would look waaay better.

tried that with search already but failed. Maybe someone can enhance that later

**Comment by ChristophWurst** [comment]
@LukasReschke ^

File: js/util/errormessagefactory.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+			}),
+			t('mail', 'We couldn’t load {tag}{name}{endtag}', {
+				name: folderName
+			}),
+			t('mail', 'There was a problem loading {tag}{name}{endtag}', {
+				name: folderName
+			})
+		];
+		var texts = _.map(rawTexts, function(text) {
+			return text.replace('{tag}', '<strong>').replace('{endtag}', '</strong>');
```
Comment (skjnldsv) [inline js/util/errormessagefactory.js]: Are you trying to bold the folder name here?

Comment (ChristophWurst) [inline js/util/errormessagefactory.js]: yes

Comment (skjnldsv) [inline js/util/errormessagefactory.js]: see https://github.com/nextcloud/mail/pull/57#issuecomment-247383309 :)
If the strong is correctly outputted on the html and still isn't displayed bold, this could be because the strong css value "font-weight" has been reset. If so just use a span and set it in css. Should be enough! ;)

Comment (jancborchardt) [inline js/util/errormessagefactory.js]: Riight, we need to add in our CSS:
`strong { font-weight: 600; }`

@skjnldsv great to see you back here by the way! :)

Comment (jancborchardt) [inline js/util/errormessagefactory.js]: Fix at https://github.com/nextcloud/server/pull/1423

Comment (skjnldsv) [inline js/util/errormessagefactory.js]: Glad to see you both! :)

File: js/views/navigation-accounts.js [hunk 2]
```diff
@@ -49,7 +49,9 @@ define(function(require) {
 				});
 			});
 
-			folder.set('active', true);
```
Comment (ChristophWurst) [inline js/views/navigation-accounts.js]: small bugfix: if the first folder fails loading, `folder` was undefined here and therefore an error was shown in the js console.

**Comment by jancborchardt** [comment]
NICE, looks very good ;D :+1:
